### PR TITLE
Fix bug where a github api failure would prevent the label check from failing

### PR DESCRIPTION
**Pull request by ZainRizvi**
Fix bug where a github api failure would prevent the check from failing even if we already saw that labels were needed.

Also adds more debugging info to the rate limit exceeded error since it's weird to see an error claiming the rate limit has exceeded when the "Used" amount is way below the limit.  I suspect these happen when the request arrived just before the rate reset time, but the response was generated right after the reset time, hence the apparently tiny "used" amounts

Example run where the check should have failed, but passed instead:
https://github.com/pytorch/pytorch/actions/runs/4200205209/jobs/7285979824

### CODE REVIEW

File: .github/scripts/check_labels.py [hunk 1]
```diff
@@ -75,16 +76,20 @@ def main() -> None:
...
+            exit_code = 1
             print(ERR_MSG)
             add_comment(pr)
-            exit(1)
         else:
             delete_comments(pr)
     except Exception as e:
         pass
 
+    exit(needs_labels)
```
Comment (huydhn) [inline .github/scripts/check_labels.py:91]: Did you mean `exit(exit_code)` here and `exit_code = 0` on line 79?

Comment (ZainRizvi) [reply .github/scripts/check_labels.py:91]: 🤦‍♂️ forgot to remove some dead code.  `exit()` converts boolean values into 0/1

**Review by huydhn — APPROVED** [review]
LGTM!

**Pull request closed without merging**